clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/handlers/BasicAmazonDeployHandler.groovy

```diff
@@ -101,21 +103,80 @@ class BasicAmazonDeployHandler implements DeployHandler<BasicAmazonDeployDescrip
+          // from a Server Group in a different region
+          if (description.classicLinkVpcId && description.classicLinkVpcSecurityGroups) {
+            def groupIds = classicLinkGroupNames.findAll { it.matches(~/sg-[0-9a-f]+/) } ?: []
+            classicLinkGroupNames.removeAll(groupIds)
+            if (groupIds) {
+              def describeSG = new DescribeSecurityGroupsRequest().withGroupIds(groupIds)
+              def provider = sourceRegionScopedProvider ?: regionScopedProvider
+              def resolvedNames = provider.amazonEC2.describeSecurityGroups(describeSG).securityGroups.findResults {
+                if (it.vpcId == classicLinkVpcId && groupIds.contains(it.groupId)) {
+                  return it.groupName
+                }
+                return null
+              } ?: []
+
+              if (resolvedNames.size() != groupIds.size()) {
+                throw new IllegalStateException("failed to look up classic link security groups, had $groupIds found $resolvedNames")
+              }
+              classicLinkGroupNames.addAll(resolvedNames)
+            }
+          }
+
+          if (deployDefaults.addAppGroupsToClassicLink) {
+            //if we cloned to a new cluster, don't bring along the old clusters groups
+            if (description.source) {
+              def srcName = Names.parseName(description.source.asgName)
+              boolean mismatch = false
+              if (srcName.app != description.application) {
+                classicLinkGroupNames.remove(srcName.app)
+                mismatch = true
+              }
+              if (srcName.stack && (mismatch || srcName.stack != description.stack)) {
+                classicLinkGroupNames.remove("${srcName.app}-${srcName.stack}".toString())
+                mismatch = true
+              }
+              if (srcName.detail && (mismatch || srcName.detail != description.freeFormDetails)) {
+                classicLinkGroupNames.remove("${srcName.app}-${srcName.stack ?: ''}-${srcName.detail}")
+              }
+            }
+            def groupNamesToLookUp = []
+            if (!classicLinkGroupNames.contains(description.application)) {
+              groupNamesToLookUp.add(description.application)
+            }
+            if (description.stack) {
+              String stackGroup = "${description.application}-${description.stack}"
+              if (!classicLinkGroupNames.contains(stackGroup)) {
+                groupNamesToLookUp.add(stackGroup)
+              }
+            }
+            if (description.freeFormDetails) {
+              String clusterGroup = "${description.application}-${description.stack ?: ''}-${description.freeFormDetails}"
+              if (!classicLinkGroupNames.contains(clusterGroup)) {
+                groupNamesToLookUp.add(clusterGroup)
+              }
+            }
+            if (groupNamesToLookUp && classicLinkGroupNames.size() < deployDefaults.maxClassicLinkSecurityGroups) {
```

Comment:
the getSecurityGroupIds call can be really slow (throttling I guess?) so that is what I was looking to avoid